Add form filters links

myhro opened the pull request:
Hi,

I always (intuitively) tried to click on the Repository/Architecture/Maintainer columns to filter the list. Links were added to them, allowing the same form filters to be applied in a simpler/quicker way.

Regards,
Tiago.

myhro committed:
Fix reCAPTCHA config check

myhro committed:
Add form filters links

myhro commented:
Updated the PR to include all `/packages`, `/contents` and `/flagged` pages.

Also fixed the reCAPTCHA config check and the `repo` filter that was not working on the Contents page.

myhro commented:
@clandmeter Hi. What are your thoughts on this? Even if you think the links are a change that isn't needed, I would still like to get the fixes for the repository filtering and the reCAPTCHA configuration check merged.

clandmeter commented:
Hi, I think its fine, but I want to first test the user experience with this patch, so please wait until I have tested it.

myhro commented:
That's great. Thanks for you feedback.

clandmeter commented:
I pushed the repo fix for contents, thx!

Regarding the filter I think its a nice feature to have, but I think its not very obvious how its implemented now. People could be confused why some items are clickable now. Some idea is to add a tooltip when users hover the link.

Recaptcha is now verified in two places, one if its enabled and one if sitekey is set. It would make more sense to switch the checks around and use the error message when no sitekey is found and recaptcha is enabled. Having recaptcha enabled with no sitekey should definitely display an error.

myhro commented:
Hi. Sorry for the delay. Only had time to look at it today.

Thanks for tooltip idea. I've updated the PR and the links purpose look much clear now.

> Recaptcha is now verified in two places, one if its enabled and one if sitekey is set. It would make more sense to switch the checks around and use the error message when no sitekey is found and recaptcha is enabled. Having recaptcha enabled with no sitekey should definitely display an error.

I realised that the error I was facing happened because `flagRenderer:post()` was checking if `conf.rc.sitekey` was set (like a `boolean`), not if it was an empty string. But indeed, it makes much more sense to first check if reCAPTCHA is enabled and only then warn if no site key was configured. I've updated these checks as well.

Can you please review the changes?

TBK commented:
@clandmeter ping. Would be great to get these features added.
